ansible: 2.11.1 -> 2.11.2; 2.10.10 ->2.10.11; 2.9.22 -> 2.9.23

mweinelt opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

New upstream release.

No idea where they keep their release notes or changelog these days. :disappointed: 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

mweinelt committed:
ansible: 2.11.1 -> 2.11.2

SuperSandro2000 commented:
This file looks promising https://github.com/ansible/ansible/blob/stable-2.11/changelogs/CHANGELOG-v2.11.rst#v2-11-2

mweinelt committed:
ansible_2_10: 2.10.10 -> 2.10.11

mweinelt committed:
ansible_2_9: 2.9.22 -> 2.9.23

mweinelt commented:
Whew, need to rework ansible-collections handling a bit.

mweinelt committed:
ansible, ansible_2_10: internalize collections package

This drops the python3Packages.ansible-collections attribute in favor of
a local callPackage that overwrites the collections package per ansible
version.

SuperSandro2000 commented on pkgs/development/python-modules/ansible/collections.nix:
```diff
@@ -14,21 +14,23 @@
 , xmltodict
 , withJunos ? false
 , withNetbox ? false
+
+, version ? null
+, sha256 ? null
```
I don't think we want the ``? null`` here. If the values are not supplied it should error.

SuperSandro2000 commented on pkgs/top-level/python-packages.nix:
```diff
@@ -441,8 +441,6 @@ in {
 
   ansible-base = callPackage ../development/python-modules/ansible/base.nix { };
 
-  ansible-collections = callPackage ../development/python-modules/ansible/collections.nix { };
```
Do we need an alias for this?

mweinelt replied:
No, ansible should be its only consumer.

mweinelt committed:
python3Packages.resolvelib: 0.7.0 -> 0.7.1

risicle commented:
`nixpkgs-review` happy, macos 10.15, nixos x86_64
